misc/json-to-record-converter/src/test/java/io/ballerina/jsonmapper/JsonToRecordMapperTests.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.jsonmapper;
+
+
+import io.ballerina.jsonmapper.diagnostic.JsonToRecordMapperDiagnostic;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests for JsonToRecordDirectConverter.
+ */
+public class JsonToRecordMapperTests {
+
+    private static final Path RES_DIR = Paths.get("src/test/resources/").toAbsolutePath();
+
+    @Test(description = "Test all sample JSON values")
+    public void testSamples() throws IOException {
```

Comment:
We better rethink about this test cases breakdown, since the current breakdown seems to be focused on the test sample types we've used. IMO we better focus on adding test cases so that we can cover all possibilities of json values that we can have.